packages/flutter/lib/src/widgets/fade_in_image.dart

```diff
@@ -454,8 +501,9 @@ class _FadeInImageState extends State<FadeInImage> with TickerProviderStateMixin
       color: new Color.fromRGBO(255, 255, 255, _animation?.value ?? 1.0),
       colorBlendMode: BlendMode.modulate,
       fit: widget.fit,
-      alignment: widget.alignment,
+      alignment: widget.alignment is FractionalOffset ? widget.alignment : widget.alignment.resolve(Directionality.of(context)),
```

Comment:
This is less efficient than doing the resolve on the `RenderImage` because if you're using a `FractionalOffsetDirectional`, you'll re-resolve every build where as the `RenderImage` pattern only re-resolves when either the `alignment` or the `textDirection` change.

Reply:
if I'm going to send a textDirection down there anyway, I guess I can move all the flipHorizontally logic there too...

Reply:
(that'll bring it from 3 sites to 2)